eval/action.yml

```diff
@@ -0,0 +1,66 @@
+name: "Sandard Eval"
+description: "Evaluate Standard based projects"
+
+inputs:
+  key_path:
+    description: "The path to place the Nix secret Key"
+    required: true
+    default: "${{ runner.temp }}/.nix-key"
+  github_pat:
+    description: "GitHub Token for Accessing Private Repos with Nix"
+    required: true
+    default: ${{ github.token }}
+  nix_key:
+    description: "Secret Key used for Nix Package Signing"
+    required: false
+  s3_key:
+    description: "Key to upload to S3 Buckets with Nix"
+    required: false
+  s3_id:
+    description: "ID to upload to S3 Buckets with Nix"
+    required: false
+  cache:
+    description: "Nix cache to optionally upload artifacts to"
+    # auto refers to the local nix store, so nothing is uploaded by default
+    default: auto
+    required: false
+
+
+runs:
+  using: "composite"
+  steps:
+    - name: Emplace Secrets
+      run: ${{ github.action_path }}/secrets.sh
+      shell: bash
+      env:
+        NIX_SECRET_KEY: ${{ inputs.nix_key }}
+        NIX_KEY_PATH: ${{ inputs.key_path }}
+        AWS_ACCESS_KEY_ID: ${{ inputs.s3_id }}
+        AWS_SECRET_ACCESS_KEY: ${{ inputs.s3_key }}
+
+    - name: Install Nix
+      uses: cachix/install-nix-action@v18
+      with:
+        extra_nix_config: |
+          accept-flake-config = true
+          trusted-users = runner
+          access-tokens = github.com=${{ inputs.github_pat }}
+          secret-key-files = ${{ inputs.key_path }}
+
+    - name: Run Eval
+      env:
+        TMPDIR: ${{ runner.temp }}
+      run: |
+        ${{ github.action_path }}/eval.sh
+      shell: bash
+
+    - name: Upload Eval Closure
+      uses: actions/upload-artifact@v3.1.1
+      with:
+        name: eval-closure
+        path: ${{ runner.temp }}/eval-closure.tar.zstd
+
+    - name: Upload Eval Closure to Binary Cache
```

Comment:
this should probably be mutually exclusive with the previous step

Reply:
Isn't it already? If the previous step fails we never get here.

Reply:
And if it doesn't fail, _should_ we get here? (Is this _really_ an obligatory comularive action?)

Reply:
what I am saying is that if the previous step fails the job just fails, it never tries to run this step at all.